plugins/outputs/yandex_cloud_monitoring/yandex_cloud_monitoring.go

```diff
@@ -0,0 +1,276 @@
+	defaultMetadataFolderUrl = "http://169.254.169.254/computeMetadata/v1/instance/attributes/folder-id"
+	maxRequestBodySize       = 4000000
+)
+
+var sampleConfig = `
+  ## Timeout for HTTP writes.
+  # timeout = "20s"
+
+  ## Normally should not be changed
+  # endpoint_url = "https://monitoring.api.cloud.yandex.net/monitoring/v2/data/write"
+
+  ## Normally folder ID is taken from Compute instance metadata
+  # folder_id = "..."
+
+  ## Can be set explicitly for authentification debugging purposes 
+  # iam_token = "..."  
+`
+
+// Description provides a description of the plugin
+func (a *YandexCloudMonitoring) Description() string {
+	return "Send aggregated metrics to Yandex.Cloud Monitoring"
+}
+
+// SampleConfig provides a sample configuration for the plugin
+func (a *YandexCloudMonitoring) SampleConfig() string {
+	return sampleConfig
+}
+
+// Connect initializes the plugin and validates connectivity
+func (a *YandexCloudMonitoring) Connect() error {
+	if a.Timeout.Duration <= 0 {
+		a.Timeout.Duration = defaultRequestTimeout
+	}
+
+	a.client = &http.Client{
+		Transport: &http.Transport{
+			Proxy: http.ProxyFromEnvironment,
+		},
+		Timeout: a.Timeout.Duration,
+	}
+
+	if a.EndpointUrl == "" {
+		a.EndpointUrl = defaultEndpointUrl
+	}
+	if a.MetadataTokenUrl == "" {
+		a.MetadataTokenUrl = defaultMetadataTokenUrl
+	}
+	if a.MetadataFolderUrl == "" {
+		a.MetadataFolderUrl = defaultMetadataFolderUrl
+	}
+	if a.FolderID == "" {
+		folderID, err := getFolderIDFromMetadata(a.client, a.MetadataFolderUrl)
+		if err != nil {
+			return err
+		}
+		a.FolderID = folderID
+	}
+	if a.Service == "" {
+		a.Service = "custom"
+	}
+
+	log.Printf("D! Writing to Yandex.Cloud Monitoring URL: %s", a.EndpointUrl)
+
+	tags := map[string]string{}
+	a.MetricOutsideWindow = selfstat.Register("yandex_cloud_monitoring", "metric_outside_window", tags)
+
+	return nil
+}
+
+// Close shuts down an any active connections
+func (a *YandexCloudMonitoring) Close() error {
+	a.client = nil
+	return nil
+}
+
+// Write writes metrics to the remote endpoint
+func (a *YandexCloudMonitoring) Write(metrics []telegraf.Metric) error {
+	var yandexCloudMonitoringMetrics []yandexCloudMonitoringMetric
+	for _, m := range metrics {
+		for _, field := range m.FieldList() {
+			yandexCloudMonitoringMetrics = append(
+				yandexCloudMonitoringMetrics,
+				yandexCloudMonitoringMetric{
+					Name:   field.Key,
+					Labels: m.Tags(),
+					TS:     fmt.Sprint(m.Time().Format(time.RFC3339)),
+					Value:  field.Value.(float64),
+				},
+			)
+		}
+	}
+
+	var body []byte
+	jsonBytes, err := json.Marshal(
+		yandexCloudMonitoringMessage{
+			Metrics: yandexCloudMonitoringMetrics,
+		},
+	)
+
+	if err != nil {
+		return err
+	}
+	// Send batches that exceed this size via separate write requests.
+	if (len(body) + len(jsonBytes) + 1) > maxRequestBodySize {
+		err := a.send(body)
+		if err != nil {
+			return err
+		}
+		body = nil
+	}
```

Comment:
body is nil here. there's no need to post it separately, and I don't see how this solves the problem where the jsonBytes is larger than maxRequestBodySize.

Reply:
We do not implement special limit for POST at server side now. So I removed this part for now.